lms/djangoapps/discussion/views.py

```diff
@@ -557,7 +604,14 @@ def followed_threads(request, course_key, user_id):
         query_params['num_pages'] = paginated_results.num_pages
         user_info = cc.User.from_django_user(request.user).to_dict()
 
-        with newrelic.agent.FunctionTrace(nr_transaction, "get_metadata_for_threads"):
+        if newrelic:
+            with newrelic.agent.FunctionTrace(nr_transaction, "get_metadata_for_threads"):
+                annotated_content_info = utils.get_metadata_for_threads(
+                    course_key,
+                    paginated_results.collection,
+                    request.user, user_info
+                )
+        else:
             annotated_content_info = utils.get_metadata_for_threads(
```

Comment:
I don't like the amount of duplication between the "if newrelic" branch and the "else" branch.  All of the "with newrelic.agent.FunctionTracer" uses here involve duplicating the body of the with.  

This feels to me like we should have the newrelic imports in a new module (perf_monitor?), and that module can provide no-op implementations for when newrelic isn't available.  That way the flow of the code doesn't need to be interrupted to deal with newrelic being missing.

This could also let us simplify the use of the module. Instead of:
```
nr_transaction = newrelic.agent.current_transaction()
with newrelic.agent.FunctionTrace(nr_transaction, "do_something"):
    do_something()
```
we could write:
```
with perf_monitor.function_trace("do_something"):
    do_something()
```